Doc/library/importlib.rst

```diff
@@ -1284,66 +1288,71 @@ find and load modules.
-   while the *values* are usually equivalent, they can differ since there is
-   no synchronization between the two objects.  Thus it is possible to update
-   the module's ``__path__`` at runtime, and this will not be automatically
-   reflected in ``__spec__.submodule_search_locations``.
+   attribute available directly on the module object. For example,
+   ``module.__spec__.name == module.__name__``.  Note however that while
+   the *values* are usually equivalent, they can differ since there is no
+   synchronization between the two objects.  For example, it is possible
+   to update the module's ``__name__`` at runtime and this will not be
+   automatically reflected in the module's ``__spec__.name``, or the
+   other way round.
 
    .. versionadded:: 3.4
 
    .. attribute:: name
 
    (``__name__``)
 
-   A string for the fully-qualified name of the module.
+   The module's fully-qualified name.
 
    .. attribute:: loader
 
    (``__loader__``)
 
-   The loader to use for loading.  For namespace packages this should be
-   set to ``None``.
+   The module's loader.
 
    .. attribute:: origin
 
    (``__file__``)
 
-   Name of the place from which the module is loaded, e.g. "builtin" for
-   built-in modules and the filename for modules loaded from source.
-   Normally "origin" should be set, but it may be ``None`` (the default)
-   which indicates it is unspecified (e.g. for namespace packages).
+   The origin of the module's source code.
+   It is ``'built-in'`` for built-in modules, ``'frozen'`` for
+   frozen modules and ``None`` for namespace packages.
 
    .. attribute:: submodule_search_locations
 
    (``__path__``)
 
-   List of strings for where to find submodules, if a package (``None``
-   otherwise).
+   The (possibly empty) iterable  of the locations of the module's
+   submodules.
+   It is ``None`` for non-package modules.
 
    .. attribute:: loader_state
 
-   Container of extra module-specific data for use during loading (or
-   ``None``).
+   The container of the module's additional data meant for use by
+   the finder and corresponding loader.
 
    .. attribute:: cached
 
    (``__cached__``)
-
-   String for where the compiled module should be stored (or ``None``).
+   The location of the module's compiled code.
+   It is ``None`` for built-in modules, frozen modules and
+   namespace packages.
 
    .. attribute:: parent
 
    (``__package__``)
 
-   (Read-only) Fully-qualified name of the package to which the module
-   belongs as a submodule (or ``None``).
+   The module's ``__name__`` attribute for a package, the parent
+   package's ``__name__`` attribute for a non-package submodule and
+   ``''`` for a non-package top-level module.
+   It is a read-only attribute.
 
    .. attribute:: has_location
 
-   Boolean indicating whether or not the module's "origin"
-   attribute refers to a loadable location.
+   The flag indicating whether or not the module's ``__origin__`` attribute
```

Comment:
The original seems fine.  If need be, I'd be more comfortable with something like the following:

```
``True`` if the loader should treat the spec's "origin" attribute as a location from which it can load the module.
```

Listing examples is fine, but it needs to be clear that (like "origin") this is up to the finder and only the loader really cares.  So listing all the default kinds of modules doesn't really add much useful information.

Reply:
Okay.

Reply:
re-visited in a follow-up review